esmvaltool/diag_scripts/weighting/climwip/core_functions.py

```diff
@@ -54,14 +54,31 @@ def distance_matrix(values: 'np.ndarray',
     return d_matrix
 
 
-def calculate_independence(data_array: 'xr.DataArray') -> 'xr.DataArray':
-    """Calculate independence.
+def calculate_model_distances(
+        data_array: 'xr.DataArray',
+        dimn: str = 'model_ensemble_reference') -> 'xr.DataArray':
+    """Calculate pair-wise distances between all values in data_array.
 
-    The independence is calculated as a distance matrix between the
-    datasets defined in the `data_array`. Returned is a square matrix
+    Distances are calculated as the area weighted eclidean distance
+    between each pair of models in data_array. Returned is a square matrix
     with where the number of elements along each edge equals the number
     of ensemble members.
+
+    Parameters
+    ----------
+    data_array : array_like, shape (N,...)
+        Array of (2 dimensional) model fields.
+    dimn : string
```

Comment:
dimn = dimension? It is clearer to write it out, imo.

```suggestion
    dimension : string
```